src/components/GridItem/Sketch.astro

```diff
@@ -1,17 +1,24 @@
 ---
 import Image from "@components/Image/index.astro";
 import {
-  makeThumbnailUrl,
+  getSketchThumbnailSource,
   thumbnailDimensions,
   type OpenProcessingCurationResponse,
 } from "@/src/api/OpenProcessing";
 
 interface Props {
   item: OpenProcessingCurationResponse[number];
+  scale?: number;
   lazyLoad?: boolean;
 }
 
-const { item, lazyLoad } = Astro.props;
+const { item, lazyLoad, scale = 1 } = Astro.props;
+const imgSrc = await getSketchThumbnailSource(item.visualID)
+const width = Math.min(thumbnailDimensions * scale, 1500);
+const height = Math.min(
+  (typeof imgSrc === 'string' ? 1 : (imgSrc as any).height / (imgSrc as any).width) * thumbnailDimensions * scale,
+  1000,
+);
```

Comment:
I think for these we can pass in `thumbnailDimensions` when `imgSrc` is a string and otherwise leave it undefined. Astro should be able to determine the right width and height based on the layout and the local image file. Or perhaps you already tried that and found an issue?

Reply:
Are you thinking of replacing `scale` with just a `width` property that you pass in? Or are you talking more about omitting the manual `width` and `height` entirely for non-OP images? (For the latter, it looks like Astro is still requesting the full 1500x1000 image for me, but maybe there's more props I'd need to pass in?)

Reply:
Ah right, I think you'd have to do something like

```astro
<Image
    src={imgSrc as any}
    alt="Screenshot of sketch"
    widths={[thumbnailDimensions, 800, 1200, 1500]}
    sizes={`(max-width: 767px) ${thumbnailDimensions}, (max-width: 1024px) 800px, (max-width: 1280px) 1200px, 1500px`}
    width={width}
    height={height}
    loading={lazyLoad ? "lazy" : "eager"}
  />
```

However, this is really just based on page load size, so it doesn't actually match the optimization you're doing based on grid position.

When I tried this out, the layout breaks between 1024px and 1280px, which is between the laptop and desktop breakpoints, so some part of the css must be depending on different intrinsic sizes for the images.

That all being said, perhaps its best to keep what you have instead of going further down this rabbit hole. :)

Reply:
I'm happy to have you merge as-is!

Reply:
Ahhh right, it would be great if it could use the size of the image itself as a breakpoint. Ok sounds good, I'll leave it as is for now!